### PR TITLE
influxctl 2.4.1

### PR DESCRIPTION
**Pull request by sanderson**
- Adds release notes for `influxctl` 2.4.0 and 2.4.1 release notes
- Documents new `influxctl write` and `influxctl query` commands
- Updates a few related page titles

**Note:** I'm going to add `influxctl write` and `influxctl query` examples to the Getting started, Write data, and Query data sections in a separate PR.

- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
+---
+title: influxctl query
+description: >
+  The `influxctl query` command command queries data from InfluxDB Cloud Dedicated
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md:4]: double command?
```suggestion
  The `influxctl query` command queries data from InfluxDB Cloud Dedicated
```

File: content/influxdb/clustered/reference/cli/influxctl/query.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
+---
+title: influxctl query
+description: >
+  The `influxctl query` command command queries data from InfluxDB Clustered
```
Comment (powersj) [inline content/influxdb/clustered/reference/cli/influxctl/query.md:4]:
```suggestion
  The `influxctl query` command queries data from InfluxDB Clustered
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+[stdin](#)
+{{% /code-tabs %}}
+{{% code-tab-content %}}
+{{% influxdb/custom-timestamps %}}
+```sh
+influxctl query \
+  --token DATABASE_TOKEN \
+  --database DATABASE_NAME \
+  "SELECT * FROM home WHERE time >= '2022-01-01T08:00:00Z' LIMIT 5"
+```
```
Comment (powersj) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md:105]: Is it worth showing one example without the token and database flags?

Comment (sanderson) [reply content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md:105]: Sure, I'll add one.

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+{{% /note %}}
+
+### InfluxDB connection configuration
+
+Your {{< product-name omit=" Clustered" >}} cluster host and port should be
+configured in your in your `influxctl`
+[connection profile](/influxdb/cloud-dedicated/reference/cli/influxctl/#configure-connection-profiles).
+By default TLS and port 443 are used.
+You can also set a default database and token to use for the `query` and `write`
+commands in your connection profile, but these can also be passed with the
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md:41]:
```suggestion
You can set a default database and token to use for the `query` and `write`
commands in your connection profile or pass them with the
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+command using the `--database` and `--token` flags.
+Command line flags override settings in the connection profile. 
+
+### Output format
+
+The `--format` option lets you print the output in other formats.
+By default, the 'table' format is used, but the 'json' format is
+available for programmatic parsing by other tooling.
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md:49]:
```suggestion
Default is 'table' format, but the 'json' format is
available for programmatic parsing by other tooling.
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/write.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  multiple lines of line protocol and will batch data based on the `--batch-size`.
+- This command is not meant to be a full, feature-rich write tool.
+  It's meant for debug, triage, and initial exploration.
+{{% /note %}}
+
+### InfluxDB connection configuration
+
+Your {{< product-name omit=" Clustered" >}} cluster host and port should be
+configured in your in your `influxctl`
+[connection profile](/influxdb/cloud-dedicated/reference/cli/influxctl/#configure-connection-profiles).
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/write.md:37]: > should be

Does this they are already or I need to go add them?

```suggestion
Your {{< product-name omit=" Clustered" >}} cluster host and port should be
configured in your `influxctl`
[connection profile](/influxdb/cloud-dedicated/reference/cli/influxctl/#configure-connection-profiles).
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/write.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+By default TLS and port 443 are used.
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/write.md:38]:
```suggestion
Default uses TLS and port 443.
```

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+- This command supports only one query per execution.
+- This command supports only SQL queries; not InfluxQL.
+- This command is not meant to be a full, feature-rich query tool.
+  It's meant for debug, triage, and basic data exploration.
+{{% /note %}}
+
+### InfluxDB connection configuration
+
+Your {{< product-name omit=" Clustered" >}} cluster host and port should be
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md:36]: > should be

They are or I need to add them?

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/write.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+You can also set a default database and token to use for the `query` and `write`
+commands in your connection profile, but these can also be passed with the
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/reference/cli/influxctl/write.md:40]:
```suggestion
You can set a default database and token to use for the `query` and `write`
commands in your connection profile or pass them with the
```

File: content/influxdb/clustered/reference/cli/influxctl/query.md [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+---
+
+The `influxctl query` command queries data from {{< product-name >}} using SQL
+and prints results as a table or JSON.
+
+Provide the SQL query in one of the following ways:
+
+- a string on the command line
+- a path to a file that contains the query
+- as a single dash (`-`) to read the query from stdin
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/query.md:23]:
```suggestion
- a single dash (`-`) to read the query from stdin
```

**Review by powersj — APPROVED** [review]
Thank you

File: content/influxdb/clustered/reference/cli/influxctl/query.md [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+- This command supports only one query per execution.
+- This command supports only SQL queries; not InfluxQL.
+- This command is not meant to be a full, feature-rich query tool.
+  It's meant for debug, triage, and basic data exploration.
+{{% /note %}}
+
+### InfluxDB connection configuration
+
+Your {{< product-name omit=" Clustered" >}} cluster host and port should be
+configured in your in your `influxctl`
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/query.md:37]: > should be

They are or I need to add them?

File: content/influxdb/clustered/reference/cli/influxctl/query.md [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+[connection profile](/influxdb/clustered/reference/cli/influxctl/#configure-connection-profiles).
+By default TLS and port 443 are used.
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/query.md:39]:
```suggestion
Default is TLS and port 443.
```

File: content/influxdb/clustered/reference/cli/influxctl/query.md [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+You can also set a default database and token to use for the `query` and `write`
+commands in your connection profile, but these can also be passed with the
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/query.md:41]:
```suggestion
You can set a default database and token to use for the `query` and `write`
commands in your connection profile or pass them with the
```

File: content/influxdb/clustered/reference/cli/influxctl/query.md [hunk 14]
```diff
@@ -0,0 +1,235 @@
...
+command using the `--database` and `--token` flags.
+Command line flags override settings in the connection profile. 
+
+### Output format
+
+The `--format` option lets you print the output in other formats.
+By default, the 'table' format is used, but the 'json' format is
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/query.md:48]:
```suggestion
Default is 'table' format, but the 'json' format is
```

File: content/influxdb/clustered/reference/cli/influxctl/write.md [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+---
+
+The `influxctl write` command writes [line protocol](/influxdb/clustered/reference/syntax/line-protocol/)
+to {{< product-name >}}.
+
+Provide line protocol in one of the following ways:
+
+- a string on the command line
+- a path to a file that contains line protocol
+- as a single dash (`-`) to read line protocol from stdin
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/write.md:22]:
```suggestion
- a single dash (`-`) to read line protocol from stdin
```

File: content/influxdb/clustered/reference/cli/influxctl/write.md [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+
+- This command supports only one write request per execution, but does support
+  multiple lines of line protocol and will batch data based on the `--batch-size`.
+- This command is not meant to be a full, feature-rich write tool.
+  It's meant for debug, triage, and initial exploration.
+{{% /note %}}
+
+### InfluxDB connection configuration
+
+Your {{< product-name omit=" Clustered" >}} cluster host and port should be
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/write.md:35]: > should be

The are or I need to add them?

File: content/influxdb/clustered/reference/cli/influxctl/write.md [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+configured in your in your `influxctl`
+[connection profile](/influxdb/clustered/reference/cli/influxctl/#configure-connection-profiles).
+By default TLS and port 443 are used.
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/write.md:38]:
```suggestion
Default uses TLS and port 443.
```

File: content/influxdb/clustered/reference/cli/influxctl/write.md [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+You can also set a default database and token to use for the `query` and `write`
+commands in your connection profile, but these can also be passed with the
```
Comment (jstirnaman) [inline content/influxdb/clustered/reference/cli/influxctl/write.md:40]:
```suggestion
You can set a default database and token to use for the `query` and `write`
commands in your connection profile or pass them with the
```

**Review by jstirnaman — APPROVED** [review]
Good stuff - just some nits.

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md [hunk 19]
```diff
@@ -299,17 +301,22 @@ If stored at a non-default location, include the `--config` flag with each
     ## Choose from "clustered" or "dedicated"
     product = "dedicated"
 
+    ## Host and port
+    ## InfluxDB hostname/IP address and port. Used for query and write subcommands.
+    ## Required for InfluxDB Clustered.
+    host = "INFLUXDB_HOST"
```
[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/cloud-dedicated/reference/cli/influxctl/_index.md:307]: Good call. Don't know why I didn't think of that 😄

File: content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md (hunk 8 above)
Comment (sanderson) [reply content/influxdb/cloud-dedicated/reference/cli/influxctl/query.md:36]: They're part of the sample config. They aren't necessarily required with Dedicated, unless you want to use these `write` and `query` commands.

**Pull request merged**